ddocs/medic-client/views/doc_summaries_by_id/map.js

```diff
@@ -10,6 +10,29 @@ function(doc) {
     return parts;
   };
 
+  var getSubject = function(doc) {
+    var subject = {};
+
+    if (doc.fields && doc.fields.hasOwnProperty('patient_uuid')) {
```

Comment:
Using `hasOwnProperty` seems overly defensive. Why not just use...

```js
if (doc.fields && doc.fields.patient_uuid) {
```

Reply:
The idea was to validate whether the property exists, since I need to identify which of the reports have an unset subject (fe: incorrect syntax SMS-s where the patient ID is not set). 
I just applied the same logic to patient_uuid in case an XML form is missing it.

Reply:
Sorry, I'm not following this comment.

> The idea was to validate whether the property exists, since I need to identify which of the reports have an unknown subject.

The example code I supplied will do that as well. The only difference as far as I can tell is mine will return true if a parent object has the `patient_uuid` property, but this object shouldn't have a parent (other than Object) so it should behave the same. What am I missing?

Reply:
I updated the map function to fallback from one value to another and added an error checker to determine if the subject is indeed required and missing.